File: outdoor-saas-be/src/main/java/com/touhuwai/service/PlanCommunityService.java
package com.touhuwai.service;

import com.touhuwai.common.PageResult;
import com.touhuwai.entity.PlanCommunity;

import java.util.List;

/**
 * 方案社区关联服务接口
 */
public interface PlanCommunityService {
    
    /**
     * 根据ID查询方案社区关联
     * @param id 关联ID
     * @return 方案社区关联信息
     */
    PlanCommunity getById(Integer id);
    
    /**
     * 根据方案ID和社区ID查询
     * @param planId 方案ID
     * @param communityId 社区ID
     * @return 方案社区关联信息
     */
    PlanCommunity getByPlanAndCommunity(Integer planId, Integer communityId);
    
    /**
     * 查询所有方案社区关联
     * @return 方案社区关联列表
     */
    List<PlanCommunity> getAll();
    
    /**
     * 分页查询方案社区关联列表
     * @param planCommunity 查询条件
     * @param pageNum 页码
     * @param pageSize 每页大小
     * @return 分页结果
     */
    PageResult<PlanCommunity> getPage(PlanCommunity planCommunity, Integer pageNum, Integer pageSize);
    
    /**
     * 根据方案ID查询关联列表
     * @param planId 方案ID
     * @return 方案社区关联列表
     */
    List<PlanCommunity> getByPlanId(Integer planId);
    
    /**
     * 根据社区ID查询关联列表
     * @param communityId 社区ID
     * @return 方案社区关联列表
     */
    List<PlanCommunity> getByCommunityId(Integer communityId);
    
    /**
     * 新增方案社区关联
     * @param planCommunity 方案社区关联信息
     * @return 影响的行数
     */
    int add(PlanCommunity planCommunity);
    
    /**
     * 批量新增方案社区关联
     * @param list 方案社区关联列表
     * @return 影响的行数
     */
    int batchAdd(List<PlanCommunity> list);
    
    /**
     * 更新方案社区关联
     * @param planCommunity 方案社区关联信息
     * @return 影响的行数
     */
    int update(PlanCommunity planCommunity);
    
    /**
     * 根据ID删除方案社区关联
     * @param id 关联ID
     * @return 影响的行数
     */
    int delete(Integer id);
    
    /**
     * 批量删除方案社区关联
     * @param ids ID列表
     * @return 影响的行数
     */
    int batchDelete(List<Integer> ids);
}

File: outdoor-saas-fe/src/services/planCommunityService.ts
/**
 * 方案社区关联服务
 */
import request from './api';
import type { PageResult } from '../types/query';

export interface PlanCommunity {
  id: number;
  planId: number;
  communityId: number;
  releaseDateBegin?: string;
  releaseDateEnd?: string;
  barrierRequiredQty?: number;
  frameRequiredQty?: number;
  plan?: {
    id: number;
    planNo: string;
    planName: string;
  };
  community?: {
    id: number;
    communityNo: string;
    buildingName: string;
  };
}

export interface PlanCommunityQueryParam {
  pageNum?: number;
  pageSize?: number;
  planId?: number;
  communityId?: number;
}

export const planCommunityService = {
  // 获取所有方案社区关联
  getAll: () => request<PlanCommunity[]>('/plan-community/list'),

  // 根据ID获取
  getById: (id: number) => request<PlanCommunity>(`/plan-community/${id}`),

  // 根据方案ID和社区ID查询
  getByPlanAndCommunity: (planId: number, communityId: number) =>
    request<PlanCommunity>(`/plan-community/query?planId=${planId}&communityId=${communityId}`),

  // 分页查询（带过滤参数）
  filterPage: (params: PlanCommunityQueryParam) =>
    request<PageResult<PlanCommunity>>('/plan-community/filter/page', {
      method: 'POST',
      body: JSON.stringify(params),
    }),

  // 根据方案ID获取
  getByPlan: (planId: number) =>
    request<PlanCommunity[]>(`/plan-community/plan/${planId}`),

  // 根据社区ID获取
  getByCommunity: (communityId: number) =>
    request<PlanCommunity[]>(`/plan-community/community/${communityId}`),

  // 创建
  create: (data: Partial<PlanCommunity>) =>
    request<PlanCommunity>('/plan-community', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  // 批量创建
  createBatch: (data: Partial<PlanCommunity>[]) =>
    request<PlanCommunity[]>('/plan-community/batch', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  // 更新
  update: (data: Partial<PlanCommunity>) =>
    request<PlanCommunity>('/plan-community', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  // 删除
  delete: (id: number) =>
    request<void>(`/plan-community/${id}`, { method: 'DELETE' }),

  // 批量删除
  deleteBatch: (ids: number[]) =>
    request<void>('/plan-community/batch', {
      method: 'DELETE',
      body: JSON.stringify(ids),
    }),
};

export default planCommunityService;
